src/server/scripts/Spells/spell_warlock.cpp

```diff
@@ -187,6 +187,34 @@ class spell_warl_create_healthstone : public SpellScript
     }
 };
 
+// 108416 - Dark Pact
+class spell_warl_dark_pact : public AuraScript
+{
+    PrepareAuraScript(spell_warl_dark_pact);
+
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellEffect({ { spellInfo->Id, EFFECT_1 }, { spellInfo->Id, EFFECT_2 } });
+    }
+
+    void CalculateAmount(AuraEffect const* /*aurEff*/, int32& amount, bool& canBeRecalculated)
+    {
+        canBeRecalculated = false;
+        if (Unit* caster = GetCaster())
+        {
+            float extraAmount = caster->SpellBaseDamageBonusDone(GetSpellInfo()->GetSchoolMask()) * 2.5f;
+            int32 absorb = caster->CountPctFromCurHealth(GetEffectInfo(EFFECT_1).CalcValue(caster));
+            caster->SetHealth(caster->GetHealth() - absorb);
+            amount = CalculatePct(absorb, GetEffectInfo(EFFECT_2).CalcValue()) + extraAmount;
```

Comment:
caster should be passed to CalcValue() in case of auras modifying it

Reply:
done